source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts

```diff
@@ -1,26 +1,38 @@
 import { ApiGen_Concepts_AcquisitionFile } from '@/models/api/generated/ApiGen_Concepts_AcquisitionFile';
 import { ApiGen_Concepts_AcquisitionFileTeam } from '@/models/api/generated/ApiGen_Concepts_AcquisitionFileTeam';
 import { formatApiPersonNames } from '@/utils/personUtils';
+import { exists, isValidId } from '@/utils/utils';
 
 export class DetailAcquisitionFile {
+  isSubFile: boolean;
   fileName?: string;
   legacyFileNumber?: string;
   assignedDate?: string;
   deliveryDate?: string;
   acquisitionPhysFileStatusTypeDescription?: string;
   acquisitionTypeDescription?: string;
+  subfileInteresttypeDescription: string | null = null;
+  otherSubfileInterestTypeDescription: string | null = null;
   regionDescription?: string;
   acquisitionTeam: DetailAcquisitionFileTeam[] = [];
 
   static fromApi(model?: ApiGen_Concepts_AcquisitionFile): DetailAcquisitionFile {
     const detail = new DetailAcquisitionFile();
+
+    detail.isSubFile =
+      exists(model.parentAcquisitionFileId) && isValidId(model.parentAcquisitionFileId);
     detail.fileName = model?.fileName ?? undefined;
     detail.legacyFileNumber = model?.legacyFileNumber ?? undefined;
     detail.assignedDate = model?.assignedDate ?? undefined;
     detail.deliveryDate = model?.deliveryDate ?? undefined;
     detail.acquisitionPhysFileStatusTypeDescription =
       model?.acquisitionPhysFileStatusTypeCode?.description ?? undefined;
     detail.acquisitionTypeDescription = model?.acquisitionTypeCode?.description ?? undefined;
+
+    if (detail.isSubFile) {
+      detail.subfileInteresttypeDescription = model.subfileInterestTypeCode.description;
```

Comment:
possible NPE here? even if the subfileInterestTypeCode shouldn't be null, I'd still guard against null here.

Reply:
(for etl'd data most likely).

Reply:
updated